x/storage/keeper/cross_app_permission.go

```diff
@@ -0,0 +1,292 @@
+				ExtraData: deletePolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: err,
+		}
+	}
+
+	return sdk.ExecuteResult{
+		Payload: types.DeletePolicyAckPackage{
+			Status:    types.StatusSuccess,
+			Id:        policy.Id.BigInt(),
+			ExtraData: deletePolicyPackage.ExtraData,
+		}.MustSerialize(),
+	}
+}
+
+func (app *PermissionApp) handleCreatePolicySynPackage(ctx sdk.Context, createPolicyPackage *types.CreatePolicySynPackage) sdk.ExecuteResult {
+	err := createPolicyPackage.ValidateBasic()
+	if err != nil {
+		return sdk.ExecuteResult{
+			Payload: types.CreatePolicyAckPackage{
+				Status:    types.StatusFail,
+				Creator:   createPolicyPackage.Operator,
+				ExtraData: createPolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: err,
+		}
+	}
+
+	var policy permtypes.Policy
+	err = json.Unmarshal(createPolicyPackage.Data, &policy)
+	if err != nil {
+		return sdk.ExecuteResult{
+			Payload: types.CreatePolicyAckPackage{
+				Status:    types.StatusFail,
+				Creator:   createPolicyPackage.Operator,
+				ExtraData: createPolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: err,
+		}
+	}
+
+	resOwner, err := app.getResourceOwner(ctx, &policy)
+	if err != nil {
+		return sdk.ExecuteResult{
+			Payload: types.CreatePolicyAckPackage{
+				Status:    types.StatusFail,
+				Creator:   createPolicyPackage.Operator,
+				ExtraData: createPolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: err,
+		}
+	}
+
+	if !createPolicyPackage.Operator.Equals(resOwner) {
+		return sdk.ExecuteResult{
+			Payload: types.CreatePolicyAckPackage{
+				Status:    types.StatusFail,
+				Creator:   createPolicyPackage.Operator,
+				ExtraData: createPolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: types.ErrAccessDenied.Wrapf(
+				"Only resource owner can put policy, operator (%s), owner(%s)",
+				createPolicyPackage.Operator.String(), resOwner.String()),
+		}
+	}
+
+	app.normalizePrincipal(ctx, policy.Principal)
+	err = app.validatePrincipal(ctx, resOwner, policy.Principal)
+	if err != nil {
+		return sdk.ExecuteResult{
+			Payload: types.CreatePolicyAckPackage{
+				Status:    types.StatusFail,
+				Creator:   createPolicyPackage.Operator,
+				ExtraData: createPolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: err,
+		}
+	}
+
+	PolicyId, err := app.permissionKeeper.PutPolicy(ctx, &policy)
+	if err != nil {
+		return sdk.ExecuteResult{
+			Payload: types.CreatePolicyAckPackage{
+				Status:    types.StatusFail,
+				Creator:   createPolicyPackage.Operator,
+				ExtraData: createPolicyPackage.ExtraData,
+			}.MustSerialize(),
+			Err: err,
+		}
+	}
+
+	return sdk.ExecuteResult{
+		Payload: types.CreatePolicyAckPackage{
+			Status:    types.StatusSuccess,
+			Id:        PolicyId.BigInt(),
+			Creator:   createPolicyPackage.Operator,
+			ExtraData: createPolicyPackage.ExtraData,
+		}.MustSerialize(),
+	}
+}
+
+func (app *PermissionApp) getResourceOwner(ctx sdk.Context, policy *permtypes.Policy) (resOwner sdk.AccAddress, err error) {
```

Comment:
can `getResourceOwner`, `normalizePrincipal`,`validatePrincipal` reuse the function in the  previous `permission` module?  
Reuse code can keep consistent with the previous logic.